godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -433,5 +428,24 @@ namespace godel_surface_detection
 
       return name;
     }
+
+    void SurfaceDetection::filterFullCloud()
+    {
+      pcl::PointCloud<pcl::PointXYZRGB>::Ptr intermediate_cloud_ptr(new pcl::PointCloud<pcl::PointXYZRGB>);
+
+      // Create Processing Cloud
+      pcl::VoxelGrid<pcl::PointXYZRGB> vox;
+      vox.setInputCloud (full_cloud_ptr_);
+      vox.setLeafSize (INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE);
+      vox.filter(*intermediate_cloud_ptr);
+
+      pcl::PassThrough<pcl::PointXYZRGB> pass;
+      pass.setInputCloud (intermediate_cloud_ptr);
+      pass.setFilterFieldName ("z");
+      pass.setFilterLimits (0.01, 1.0);
```

Comment:
I understand we're moving fast to get stuff working but we may want to make this a parameterized item. At least make it a named constant.